parallaxinc/solo
# Change number block ranges to objects

Addresses #111

Changes the method that feeds the **math_number** block's range-related warnings and text to an object that is stored in the input instead of a hidden field.  This also refactors the **math_number** block's onchange event and filters the onchange to only run when the block itself is affected.  This should significantly improve BlocklyProp's performance, as the **math_number** block is 1) very common, 2) now running far less code overall, and 3) Removes over 200 lines of code!!! Yay!

To test - 
1) The following blocks are affected:
color_value_from
make_pin_input
check_pin_input
base_freqout
eeprom_write
eeprom_read
servo_move
servo_set_ramp
fb360_setup
fb360_set
ab_volt_out
pwm_set
sound_play
wav_volume
ab_drive_goto
ab_drive_goto_max_speed
ab_drive_speed
heb_pwm_led
scribbler_simple_wait
scribbler_move_motors
scribbler_move_motors_distance
scribbler_move_motors_xy
scribbler_move_motors_angle
scribbler_play_polyphony
scribbler_stop_servo
scribbler_eeprom_read
scribbler_eeprom_write
sensor_ping

2) Try various numbers in the **math_number** block to elicit the out-of-range response.
3) move the **math_number** block in and out of other block's inputs to make sure the warning text stays specific to the input (and the warnings disappear when the block is removed from the input)
